react/InfosCarrousel/index.jsx

```diff
@@ -15,6 +15,8 @@ const InfosCarrousel = ({ children, theme, className, swipeableProps }) => {
   const hasPreviousInfos = index === 0
   const hasNextInfos = index === count - 1
 
+  if (index >= count) setIndex(count - 1) // can happen when children are removed after navigating
+
```

Comment:
I am not really comfortable with setting state inside the render.

I think it would be clearer if the index could *never* go below 0 or higher than the index.

What do you think of the following approach where the value is completely constrained by a hook. No double setState are necessary, and internal state is never wrong.

```
const clamp = (value, min, max) => Math.max(min, Math.min(value, max)))

const useClampedCounter = (start, min, max) => {
  const [counter, setCounterRaw] = useState(start) 
  // maybe a useCallback to use here ?
  const setCounter = (index) => setCounterRaw(clamp(value, min, max))
  return [counter, setCounter]
}

const [index, setIndex] = useClampedCounter(0
```

Reply:
Very good point and elegant solution, I'll give it a try!

Reply:
useClampedValue instead of useClampedCounter since no specific methods like increment or decrement are returned by the hook.

Reply:
I renamed a few other things, and also didn't use `useCallback` on `setCounter` since the function is not directly passed to child components. But it works great, thanks!